src/main/io/spektrum_vtx_control.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "platform.h"
+#if defined(USE_SPEKTRUM_VTX_CONTROL) && defined(USE_VTX_COMMON)
+
+#include <string.h>
+
+#include "fc/config.h"
+#include "drivers/vtx_common.h"
+#include "io/vtx.h"
+
+#include "io/spektrum_vtx_control.h"
+
+// We can not use the common set/get-frequncy API.
+// Some VTX devices do not support it.
+//#define USE_VTX_COMMON_FREQ_API
+
+#ifdef USE_VTX_COMMON_FREQ_API
+const uint16_t SpektrumVtxfrequencyTable[SPEKTRUM_VTX_BAND_COUNT][SPEKTRUM_VTX_CHAN_COUNT] =
+    {
+        { 5740, 5760, 5780, 5800, 5820, 5840, 5860, 5880 }, // FatShark
+        { 5658, 5695, 5732, 5769, 5806, 5843, 5880, 5917 }, // RaceBand
+        { 5705, 5685, 5665, 5645, 5885, 5905, 5925, 5945 }, // Boscam E
+        { 5733, 5752, 5771, 5790, 5809, 5828, 5847, 5866 }, // Boscam B
+        { 5865, 5845, 5825, 5805, 5785, 5765, 5745, 5725 }, // Boscam A
+    };
+#else
+// Translation table, Spektrum bands to BF internal vtx_common bands
+const uint8_t spek2commonBand[SPEKTRUM_VTX_BAND_COUNT]= {
+    VTX_COMMON_BAND_FS,
+    VTX_COMMON_BAND_RACE,
+    VTX_COMMON_BAND_E,
+    VTX_COMMON_BAND_B,
+    VTX_COMMON_BAND_A,
+};
+#endif
+
+// RF Power Index translation tables. No generic power API available.....
+
+#ifdef USE_VTX_TRAMP
```

Comment:
Please don't use VTX-protocol specific tables. Use unified power level API and note that power level is dependent on the driver and actual hardware connected.